### PR TITLE
Update dry transaction docs

### PR DESCRIPTION
**Pull request by timriley**
Make a number of adjustments:

* Use a dry-container in all of the example code, so no one expects that a simple hash is reasonable everyday usage
* Use `step` steps in most of the example code, to encourage users to write operations that return `Either` objects
* Move the discussion about the other step adapters into its own section
* Add a section about the new custom step adapters

I'll be releasing these tomorrow along with the next gem release of dry-transaction.

### CODE REVIEW

File: source/gems/dry-transaction/usage.html.md [hunk 1]
```diff
@@ -129,18 +145,60 @@ NOTIFICATIONS
...
+```ruby
+class MyStepAdapters < Dry::Transaction::StepAdapters
+  register :custom_adapter, MyAdapter.new
+end
+
+save_user = Dry.Transaction(container: Container, step_adapters: MyStepAdapters) do
+  # ...
+end
+```
+
```
Comment (timriley) [inline source/gems/dry-transaction/usage.html.md]: @solnic @AMHOL – do these 2 dry-container examples seem fair? (i.e. adding an adapter globally by registering it on the existing container, or subclassing it if you want your own separate container object for your custom adapters)

**Comment by solnic** [comment]
would be nice to break that down into sub-sections. otherwise it looks really good :)

**Comment by timriley** [comment]
@solnic thanks for checking it out :) I was wondering when I might need sub-sections, and your comment has put me over the line. I'll break it apart tomorrow morning before publishing!

**Comment by timriley** [comment]
dry-transaction v0.6.0 is out, and I've arranged the docs into multiple pages now. Merging!

**Pull request merged**